### DISCUSSION TITLE
Order of execution for lock files and postUpgradeTasks

### DISCUSSION THREAD

**Discussion by jackson-chris** [renovatebot/renovate#29172] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

github, latest full image

### Please tell us more about your question or problem

If I wanted to make sure two commands were run prior to renovatebot attempting to update lock files how can I achieve that? I thought to use `postUpgradeTasks` but I am uncertain about the order of execution. Is there a document that outlines the order in which renovate will run things like: update package manager file, update lock files, postUpgradeTasks, Commit, etc.

Thanks in advance.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
You cannot. It would need the equivalent of a "preUpgradeTasks", which is an open feature request

**Reply by jackson-chris** [reply]
Thank you for the quick reply.
